### DISCUSSION TITLE
Snapcast source as radio  (music provider)

### DISCUSSION THREAD

**Discussion by frafall** [music-assistant/support#3380] [Feature requests and ideas]
Would it be feasible to have Snapcast as a music provider, ie not only as a player?

I have a number of streaming integrations like bluetooth from cell phones, record player, mpd, spotify connect  and  air play that currently are streaming into my snapcast server. 

I've looked into the python snapcast.client which hasn't really been touched for some years but it seems possible to 
have MA define a 'virtual' snapcast client to listen to a stream.

So we could have each (or selected) streams as radio stations in MA, each with a virtual snapcast client for 
audio input into  MA. 

Best regards
Morten

This would enable input through alsa, named pipes, tcp, ....

**Comment by OzGav** [comment]
Many things are possible however the current roadmap is to support a number of your sources directly. That is, Spotify Connect, AirPlay and local line in

**Reply by frafall** [reply]
Thank for reply, do you have a link to the roadmap ?

**Reply by OzGav** [reply]
Here you go https://github.com/orgs/music-assistant/projects/2
